lib/dataContract/stateTransition/validation/validateDataContractSTDataFactory.js

```diff
@@ -40,6 +45,35 @@ function validateDataContractSTDataFactory(dataProvider, validateIdentityExisten
       );
     }
 
+    if (!result.isValid()) {
+      return result;
+    }
+
+    // Verify ST signature
+    const identity = await dataProvider.fetchIdentity(dataContractId);
+    const publicKey = identity.getPublicKeyById(stateTransition.getSignaturePublicKeyId());
+    if (publicKey.getType() !== IdentityPublicKey.TYPES.ECDSA_SECP256K1) {
+      result.addError(
+        new InvalidIdentityPublicKeyType(publicKey.getType()),
+      );
+
+      return result;
+    }
+
+    let signatureIsValid;
+
+    try {
+      signatureIsValid = stateTransition.verifySignature(publicKey);
+    } catch (e) {
+      result.addError(new SignatureVerificationError(stateTransition.getSignature(), e));
```

Comment:
Let's catch dashcore-lib errors inside `verifySignature ` and return false in case of error.